elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/PermissionExpressionVisitor.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.modelconfig.validator;
+
+import com.yahoo.elide.core.security.permissions.expressions.AndExpression;
+import com.yahoo.elide.core.security.permissions.expressions.Expression;
+import com.yahoo.elide.core.security.permissions.expressions.NotExpression;
+import com.yahoo.elide.core.security.permissions.expressions.OrExpression;
+import com.yahoo.elide.generated.parsers.ExpressionBaseVisitor;
+import com.yahoo.elide.generated.parsers.ExpressionParser;
+
+import lombok.AllArgsConstructor;
+
+import java.util.function.Function;
+
+/**
+ * Expression Visitor.
+ */
+@AllArgsConstructor
+public class PermissionExpressionVisitor extends ExpressionBaseVisitor<Expression> {
```

Comment:
The visitor doesn't have to return an expression.  The visitor can construct anything really.  I thought you might construct a `Set<String>` of permission rules - that then could be iterated over and validated against the roles or existing security checks.

Reply:
Updated.